### PR TITLE
Make :logger_file_backend an opt-in dependency

### PR DESCRIPTION
**Pull request by oshosanya**
- Made :logger_file_backend an opt-in-dependency
- Only configure `LoggerFileBackend` if module is loaded

### CODE REVIEW

File: lib/crawly/engine.ex [hunk 1]
```diff
@@ -61,8 +61,15 @@ defmodule Crawly.Engine do
...
-    if Crawly.Utils.get_settings(:log_to_file, spider_name) do
-      configure_spider_logs(spider_name, opts[:crawl_id])
+    case {Crawly.Utils.get_settings(:log_to_file, spider_name),
+          Code.ensure_loaded?(LoggerFileBackend)} do
+      {true, true} ->
+        configure_spider_logs(spider_name, opts[:crawl_id])
+
+      {true, false} ->
+        Logger.warn(
+          ":logger_file_backend must be installed as a peer dependency if log_to_file config is set to true"
```
Comment (Ziinc) [inline lib/crawly/engine.ex:71]: Can you add in a link to the LoggerFileBackend docs?

File: documentation/configuration.md [hunk 2]
```diff
@@ -139,7 +139,7 @@ stored under the `log_dir` folder. This option is ignored if `log_to_file` is no
 
 default: false
 
-Enables or disables file logging.
+Enables or disables file logging. If set to `true`, make sure to add `:logger_file_backend` as a dependency to your project.
```
Comment (Ziinc) [inline documentation/configuration.md:142]: Can you add in a link to the LoggerFileBackend docs?

File: lib/crawly/engine.ex (hunk 1 above)
Comment (Ziinc) [reply lib/crawly/engine.ex:71]: for better ux.

**Review by Ziinc — CHANGES_REQUESTED** [review]
could you add in some tests for different situations where the return values of `Code.ensure_loaded?/1`?

For example:
- if false, should not call the LoggerFileBackend. Can use :meck to check if it is called.
- if true, should call it. can use meck.


you can adapt the existing test that i added in the previous PR

**Comment by Ziinc** [comment]
You'll also need to resolve the failing  tests as shown in circle ci.

File: lib/crawly/engine.ex [hunk 3]
```diff
@@ -61,8 +61,15 @@ defmodule Crawly.Engine do
       |> Map.put_new_lazy(:crawl_id, &UUID.uuid1/0)
 
     # Filter all logs related to a given spider
-    if Crawly.Utils.get_settings(:log_to_file, spider_name) do
-      configure_spider_logs(spider_name, opts[:crawl_id])
+    case {Crawly.Utils.get_settings(:log_to_file, spider_name),
```
[the remark this answers is not in the document]
Comment (oshosanya) [reply lib/crawly/engine.ex:64]: Done

File: lib/crawly/engine.ex (hunk 1 above)
Comment (oshosanya) [reply lib/crawly/engine.ex]: Done

File: documentation/configuration.md (hunk 2 above)
Comment (oshosanya) [reply documentation/configuration.md]: Done

**Review by Ziinc — APPROVED** [review]
looks good, thanks for the PR 😄

**Comment by oshosanya** [comment]
> You'll also need to resolve the failing tests as shown in circle ci.

If I can run the tests again, it should pass because of the inconsistency I had spoken to you about. Currently, tests on Circle CI can not be rerun

**Comment by Ziinc** [comment]
> > You'll also need to resolve the failing tests as shown in circle ci.
> 
> If I can run the tests again, it should pass because of the inconsistency I had spoken to you about. Currently, tests on Circle CI can not be rerun

I had already manually triggered the re-run when i approved the changes. Some tests are a little flaky and affecting other PRs as well. Will need to look into them.

**Pull request merged**